packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -109,6 +109,65 @@ export interface ElementScreenshotOptions extends ScreenshotOptions {
   scrollIntoView?: boolean;
 }
 
+/**
+ * A given method will have it's `this` replaced with an isolated version of
+ * `this` when decorated with this decorator.
+ *
+ * All changes of isolated `this` are reflected on the actual `this`.
+ *
+ * @internal
+ */
+export function bindIsolatedHandle<This extends ElementHandle<Node>>(
```

Comment:
since we are moving it, can we move it to the decorators.ts if that works?

Reply:
It should work it's a function and the types should allow. Let me try.

Reply:
We can't move it due to the use of `instance of JsHandle` as it created a cycle dependencies.